cloudify_cli/commands/replace_certificates.py

```diff
@@ -0,0 +1,273 @@
+    # TODO: Take care of the AIO case
+    output_path = output_path if output_path else CERTS_CONFIG_PATH
+    config = _get_cluster_configuration_dict(client)
+    with open(output_path, 'w') as output_file:
+        yaml.dump(config, output_file, default_flow_style=False)
+
+    logger.info('The certificates replacement configuration file was '
+                'saved to {0}'.format(output_path))
+
+
+@replace_certificates.command(name='start',
+                              short_help='Replace certificates after updating '
+                                         'the configuration file')
+@cfy.options.input_path(help='The certificates replacement configuration file')
+@cfy.options.force('Use the force flag in case you want to change only a '
+                   'CA and not the certificates signed by it')
+@cfy.assert_manager_active()
+@cfy.pass_logger
+def start_replace_certificates(input_path,
+                               force,
+                               logger):
+    # TODO: implement for the AIO case
+    # TODO: Validate the user role = admin
+    _validate_username_and_private_key()
+    config_dict = get_dict_from_yaml(_get_input_path(input_path))
+    logger.info('Validating replace-certificates config file...')
+    errors_list = validate_config_dict(config_dict, force, logger)
+    if errors_list:
+        raise_errors_list(errors_list, logger)
+
+    main_config = ReplaceCertificatesConfig(config_dict, False, logger)
+    are_certs_valid = main_config.validate_certificates()
+    if are_certs_valid:
+        logger.info('Replacing certificates')
+        replace_certs_succeeded = main_config.replace_certificates()
+        new_cli_cert = main_config.new_cli_ca_cert()
+        if replace_certs_succeeded and new_cli_cert:
+            os.system('cfy profiles set -c {0}'.format(new_cli_cert))
+
+
+def _get_input_path(input_path):
+    input_path = input_path if input_path else CERTS_CONFIG_PATH
+    if not os.path.exists(input_path):
+        raise CloudifyCliError('Please create the replace-certificates '
+                               'configuration file first using the command'
+                               ' `cfy replace-certificates generate-file`')
+    return input_path
+
+
+def _get_cluster_configuration_dict(client):
+    instances_ips = _get_instances_ips(client)
+    config = {}
+    _basic_config_update(config)
+    _add_manager_nodes_to_config_instance(config,
+                                          instances_ips['manager_nodes_ips'])
+    _add_nodes_to_config_instance(config, 'postgresql_server',
+                                  instances_ips['db_nodes_ips'])
+    _add_nodes_to_config_instance(config, 'rabbitmq',
+                                  instances_ips['broker_nodes_ips'])
+    return config
+
+
+def _add_manager_nodes_to_config_instance(config, manager_ips):
+    for host_ip in manager_ips:
+        instance = {'host_ip': str(host_ip),
+                    'new_internal_cert_path': '',
```

Comment:
ALL of them (here and certs_mapping etc) start with `new` and end with `path`, so i'd just drop those pre/postfixes and named it just `internal_cert` instead. Makes it a bit easier on the eyes when reading.

Reply:
Makes a lot of sense, I dropped the `new` and kept the `path` to be consistent with the config.yaml file